internal/cli/install/install.go

```diff
@@ -39,75 +44,136 @@ func (o *InstallOptions) AddFlags(cmd *cobra.Command) {
+		// update Components to default config
+		o.Components = defaultComponents
 		return nil
 	}
 	name := args[0]
 	if v, ok := o.Components[name]; ok {
-		if o.version != "" {
-			o.Components = map[string]string{name: o.version}
-		} else {
+		if o.version == "" {
 			o.Components = map[string]string{name: v}
+		} else {
+			o.Components = map[string]string{name: o.version}
 		}
 		return nil
 	}
-	return fmt.Errorf("%s install not supported", name)
+	return fmt.Errorf("component `%v` is not supported", name)
 }
 
 // Install component
-func Install(component string, version string) error {
-	var url string
+func (o *InstallOptions) Install(component, version string) error {
+	cmd, err := o.buildCmd(component, version)
+	if err != nil {
+		return err
+	}
+	return runCmd(cmd)
+}
+
+// buildCmd build cmd for installation
+func (o *InstallOptions) buildCmd(component, version string) (*exec.Cmd, error) {
 	var cmd *exec.Cmd
-	obUrl := "https://raw.githubusercontent.com/oceanbase/ob-operator/"
-	localPathUrl := "https://raw.githubusercontent.com/rancher/local-path-provisioner/"
+	var url string
 	switch component {
 	case "cert-manager":
 		componentFile := "cert-manager.yaml"
-		url = fmt.Sprintf("%s%s/deploy/%s", obUrl, version, componentFile)
+		url = fmt.Sprintf("%s%s/deploy/%s", o.obUrl, version, componentFile)
 		cmd = exec.Command("kubectl", "apply", "-f", url)
 	case "ob-operator", "ob-operator-dev":
 		componentFile := "operator.yaml"
-		url = fmt.Sprintf("%s%s/deploy/%s", obUrl, version, componentFile)
+		url = fmt.Sprintf("%s%s/deploy/%s", o.obUrl, version, componentFile)
 		cmd = exec.Command("kubectl", "apply", "-f", url)
 	case "local-path-provisioner", "local-path-provisioner-dev":
 		componentFile := "local-path-storage.yaml"
-		url = fmt.Sprintf("%s%s/deploy/%s", localPathUrl, version, componentFile)
+		url = fmt.Sprintf("%s%s/deploy/%s", o.localPathUrl, version, componentFile)
 		cmd = exec.Command("kubectl", "apply", "-f", url)
 	case "ob-dashboard":
-		if err := preRunCmd(); err != nil {
-			return err
+		if err := addHelmRepo(); err != nil {
+			return nil, err
+		}
+		if err := updateHelmRepo(); err != nil {
+			return nil, err
 		}
 		versionFlag := fmt.Sprintf("--version=%s", version)
 		cmd = exec.Command("helm", "install", "oceanbase-dashboard", "ob-operator/oceanbase-dashboard", versionFlag)
+	default:
+		return nil, fmt.Errorf("unknown component: %s", component)
 	}
-	return runCmd(cmd)
+	return cmd, nil
 }
 
-// preRunCmd preRun two commands for ob-dashboard installation
-func preRunCmd() error {
-	// add helm repo
+// checkCertManager checks cert-manager in the environment
+func checkCertManager() bool {
+	cmd := exec.Command("kubectl", "get", "crds", "-o", "name", "|", "grep", "cert-manager")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return false
+	}
+
+	// Check if the output contains cert-manager resources
+	expectedResources := []string{
```

Comment:
What if a different version of cert-manager has different CRDs?

Reply:
Fine，maybe I still need  to  run `kubectl api-resources | grep cert-manager.io/v1` to check the environment?

Reply:
Actually what I mean is that you require all CRDs in the list appear in the output of kubectl command. Would that be too strict?

Reply:
In their official documentation:
`kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.15.3/cert-manager.yaml
`should be run to install all the CRDs.
I think if the user installs this way, maybe all CRDs should be found in the environment?

Reply:
If the version is fixed to v1.15.3, the requirement is proper. How about other version like v1.14, v1.13 or upcoming v1.16? The list of CRDs should be check if they are the same between versions.

Reply:
I have tried official repo with version of v1.14, v1.13, v1.12 and v1.11, the list of CRDs are the same.

Reply:
That's great!